File: src/01-easy/ascii-art/solution.m
#import <stdio.h>
#import <string.h>
#import <ctype.h>
int main(){
    int L,H;char T[300],rows[30][1200];
    scanf("%d%d ",&L,&H);fgets(T,sizeof(T),stdin);T[strcspn(T,"\n")]=0;
    for(int i=0;i<H;i++){fgets(rows[i],sizeof(rows[i]),stdin);rows[i][strcspn(rows[i],"\n")]=0;}
    for(int i=0;i<H;i++){
        for(int j=0;T[j];j++){
            int idx=toupper(T[j])-'A';if(idx<0||idx>25)idx=26;
            printf("%.*s",L,rows[i]+idx*L);
        }
        printf("\n");
    }
}

File: src/01-easy/ascii-art/solution.go
package main
import("bufio";"fmt";"os";"strings")
func main(){
    r:=bufio.NewReader(os.Stdin)
    var L,H int;fmt.Fscan(r,&L);fmt.Fscan(r,&H);r.ReadString('\n')
    T,_:=r.ReadString('\n');T=strings.TrimSpace(strings.ToUpper(T))
    rows:=make([]string,H)
    for i:=0;i<H;i++{s,_:=r.ReadString('\n');rows[i]=strings.TrimRight(s,"\n")}
    for i:=0;i<H;i++{
        for _,c:=range T{
            idx:=int(c)-65;if idx<0||idx>25{idx=26}
            fmt.Print(rows[i][idx*L:idx*L+L])
        }
        fmt.Println()
    }
}
